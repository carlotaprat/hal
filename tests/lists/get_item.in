def map! list:
  for i in range 0, length list:
  	list[i] = yield list[i]
  list

a = [1,2,3,4,(f 4, 3*2 + e), e + 2]

a = map! a as element:
  element*2
