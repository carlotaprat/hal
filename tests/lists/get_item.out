digraph {

	ordering=out;
	ranksep=.4;
	bgcolor="lightgrey"; node [shape=box, fixedsize=false, fontsize=12, fontname="Helvetica-bold", fontcolor="blue"
		width=.25, height=.25, color="black", fillcolor="white", style="filled, solid, bold"];
	edge [arrowsize=.5, color="black", style="bold"]

  n0 [label="BLOCK"];
  n1 [label="FUNDEF"];
  n1 [label="FUNDEF"];
  n2 [label="map!"];
  n3 [label="PARAMS"];
  n3 [label="PARAMS"];
  n4 [label="list"];
  n5 [label="BLOCK"];
  n5 [label="BLOCK"];
  n6 [label="FOR_STMT"];
  n6 [label="FOR_STMT"];
  n7 [label="PARAMS"];
  n7 [label="PARAMS"];
  n8 [label="i"];
  n9 [label="FUNCALL"];
  n9 [label="FUNCALL"];
  n10 [label="range"];
  n11 [label="ARGS"];
  n11 [label="ARGS"];
  n12 [label="0"];
  n13 [label="FUNCALL"];
  n13 [label="FUNCALL"];
  n14 [label="length"];
  n15 [label="ARGS"];
  n15 [label="ARGS"];
  n16 [label="FUNCALL"];
  n16 [label="FUNCALL"];
  n17 [label="list"];
  n18 [label="ARGS"];
  n19 [label="BLOCK"];
  n19 [label="BLOCK"];
  n20 [label="="];
  n20 [label="="];
  n21 [label="[]"];
  n21 [label="[]"];
  n22 [label="FUNCALL"];
  n22 [label="FUNCALL"];
  n23 [label="list"];
  n24 [label="ARGS"];
  n25 [label="FUNCALL"];
  n25 [label="FUNCALL"];
  n26 [label="i"];
  n27 [label="ARGS"];
  n28 [label="YIELD"];
  n28 [label="YIELD"];
  n29 [label="[]"];
  n29 [label="[]"];
  n30 [label="FUNCALL"];
  n30 [label="FUNCALL"];
  n31 [label="list"];
  n32 [label="ARGS"];
  n33 [label="FUNCALL"];
  n33 [label="FUNCALL"];
  n34 [label="i"];
  n35 [label="ARGS"];
  n36 [label="RETURN"];
  n36 [label="RETURN"];
  n37 [label="FUNCALL"];
  n37 [label="FUNCALL"];
  n38 [label="list"];
  n39 [label="ARGS"];
  n40 [label="="];
  n40 [label="="];
  n41 [label="FUNCALL"];
  n41 [label="FUNCALL"];
  n42 [label="a"];
  n43 [label="ARGS"];
  n44 [label="LIST"];
  n44 [label="LIST"];
  n45 [label="1"];
  n46 [label="2"];
  n47 [label="3"];
  n48 [label="4"];
  n49 [label="FUNCALL"];
  n49 [label="FUNCALL"];
  n50 [label="f"];
  n51 [label="ARGS"];
  n51 [label="ARGS"];
  n52 [label="4"];
  n53 [label="+"];
  n53 [label="+"];
  n54 [label="*"];
  n54 [label="*"];
  n55 [label="3"];
  n56 [label="2"];
  n57 [label="FUNCALL"];
  n57 [label="FUNCALL"];
  n58 [label="e"];
  n59 [label="ARGS"];
  n60 [label="+"];
  n60 [label="+"];
  n61 [label="FUNCALL"];
  n61 [label="FUNCALL"];
  n62 [label="e"];
  n63 [label="ARGS"];
  n64 [label="2"];
  n65 [label="="];
  n65 [label="="];
  n66 [label="FUNCALL"];
  n66 [label="FUNCALL"];
  n67 [label="a"];
  n68 [label="ARGS"];
  n69 [label="FUNCALL"];
  n69 [label="FUNCALL"];
  n70 [label="map!"];
  n71 [label="ARGS"];
  n71 [label="ARGS"];
  n72 [label="FUNCALL"];
  n72 [label="FUNCALL"];
  n73 [label="a"];
  n74 [label="ARGS"];
  n75 [label="LAMBDA"];
  n75 [label="LAMBDA"];
  n76 [label="PARAMS"];
  n76 [label="PARAMS"];
  n77 [label="element"];
  n78 [label="BLOCK"];
  n78 [label="BLOCK"];
  n79 [label="*"];
  n79 [label="*"];
  n80 [label="FUNCALL"];
  n80 [label="FUNCALL"];
  n81 [label="element"];
  n82 [label="ARGS"];
  n83 [label="2"];

  n0 -> n1 // "BLOCK" -> "FUNDEF"
  n1 -> n2 // "FUNDEF" -> "map!"
  n1 -> n3 // "FUNDEF" -> "PARAMS"
  n3 -> n4 // "PARAMS" -> "list"
  n1 -> n5 // "FUNDEF" -> "BLOCK"
  n5 -> n6 // "BLOCK" -> "FOR_STMT"
  n6 -> n7 // "FOR_STMT" -> "PARAMS"
  n7 -> n8 // "PARAMS" -> "i"
  n6 -> n9 // "FOR_STMT" -> "FUNCALL"
  n9 -> n10 // "FUNCALL" -> "range"
  n9 -> n11 // "FUNCALL" -> "ARGS"
  n11 -> n12 // "ARGS" -> "0"
  n11 -> n13 // "ARGS" -> "FUNCALL"
  n13 -> n14 // "FUNCALL" -> "length"
  n13 -> n15 // "FUNCALL" -> "ARGS"
  n15 -> n16 // "ARGS" -> "FUNCALL"
  n16 -> n17 // "FUNCALL" -> "list"
  n16 -> n18 // "FUNCALL" -> "ARGS"
  n6 -> n19 // "FOR_STMT" -> "BLOCK"
  n19 -> n20 // "BLOCK" -> "="
  n20 -> n21 // "=" -> "[]"
  n21 -> n22 // "[]" -> "FUNCALL"
  n22 -> n23 // "FUNCALL" -> "list"
  n22 -> n24 // "FUNCALL" -> "ARGS"
  n21 -> n25 // "[]" -> "FUNCALL"
  n25 -> n26 // "FUNCALL" -> "i"
  n25 -> n27 // "FUNCALL" -> "ARGS"
  n20 -> n28 // "=" -> "YIELD"
  n28 -> n29 // "YIELD" -> "[]"
  n29 -> n30 // "[]" -> "FUNCALL"
  n30 -> n31 // "FUNCALL" -> "list"
  n30 -> n32 // "FUNCALL" -> "ARGS"
  n29 -> n33 // "[]" -> "FUNCALL"
  n33 -> n34 // "FUNCALL" -> "i"
  n33 -> n35 // "FUNCALL" -> "ARGS"
  n5 -> n36 // "BLOCK" -> "RETURN"
  n36 -> n37 // "RETURN" -> "FUNCALL"
  n37 -> n38 // "FUNCALL" -> "list"
  n37 -> n39 // "FUNCALL" -> "ARGS"
  n0 -> n40 // "BLOCK" -> "="
  n40 -> n41 // "=" -> "FUNCALL"
  n41 -> n42 // "FUNCALL" -> "a"
  n41 -> n43 // "FUNCALL" -> "ARGS"
  n40 -> n44 // "=" -> "LIST"
  n44 -> n45 // "LIST" -> "1"
  n44 -> n46 // "LIST" -> "2"
  n44 -> n47 // "LIST" -> "3"
  n44 -> n48 // "LIST" -> "4"
  n44 -> n49 // "LIST" -> "FUNCALL"
  n49 -> n50 // "FUNCALL" -> "f"
  n49 -> n51 // "FUNCALL" -> "ARGS"
  n51 -> n52 // "ARGS" -> "4"
  n51 -> n53 // "ARGS" -> "+"
  n53 -> n54 // "+" -> "*"
  n54 -> n55 // "*" -> "3"
  n54 -> n56 // "*" -> "2"
  n53 -> n57 // "+" -> "FUNCALL"
  n57 -> n58 // "FUNCALL" -> "e"
  n57 -> n59 // "FUNCALL" -> "ARGS"
  n44 -> n60 // "LIST" -> "+"
  n60 -> n61 // "+" -> "FUNCALL"
  n61 -> n62 // "FUNCALL" -> "e"
  n61 -> n63 // "FUNCALL" -> "ARGS"
  n60 -> n64 // "+" -> "2"
  n0 -> n65 // "BLOCK" -> "="
  n65 -> n66 // "=" -> "FUNCALL"
  n66 -> n67 // "FUNCALL" -> "a"
  n66 -> n68 // "FUNCALL" -> "ARGS"
  n65 -> n69 // "=" -> "FUNCALL"
  n69 -> n70 // "FUNCALL" -> "map!"
  n69 -> n71 // "FUNCALL" -> "ARGS"
  n71 -> n72 // "ARGS" -> "FUNCALL"
  n72 -> n73 // "FUNCALL" -> "a"
  n72 -> n74 // "FUNCALL" -> "ARGS"
  n69 -> n75 // "FUNCALL" -> "LAMBDA"
  n75 -> n76 // "LAMBDA" -> "PARAMS"
  n76 -> n77 // "PARAMS" -> "element"
  n75 -> n78 // "LAMBDA" -> "BLOCK"
  n78 -> n79 // "BLOCK" -> "*"
  n79 -> n80 // "*" -> "FUNCALL"
  n80 -> n81 // "FUNCALL" -> "element"
  n80 -> n82 // "FUNCALL" -> "ARGS"
  n79 -> n83 // "*" -> "2"

}

